lib/IDE/Refactoring.cpp

```diff
@@ -5310,6 +5337,293 @@ class AsyncConverter : private SourceEntityWalker {
     }
   }
 };
+
+/// When adding an async alternative method for the function declaration \c FD,
+/// this class tries to create a function body for the legacy function (the one
+/// with a completion handler), which calls the newly converted async function.
+/// There are certain situations in which we fail to create such a body, e.g.
+/// if the completion handler has the signature `(String, Error?) -> Void` in
+/// which case we can't synthesize the result of type \c String in the error
+/// case.
+class LegacyAlternativeBodyCreator {
+  /// The old function declaration for which an async alternative has been added
+  /// and whose body shall be rewritten to call the newly added async
+  /// alternative.
+  FuncDecl *FD;
+
+  /// The description of the completion handler in the old function declaration.
+  AsyncHandlerDesc HandlerDesc;
+
+  /// Returns the call to the refactored 'async' method without the 'await'
+  /// keyword.
+  std::string getCallToAsyncMethod() const {
```

Comment:
I suppose you mean that `LegacyAlternativeBodyCreator` should have an output stream and all methods append to it like `AsyncConverter` does.

I was using that initially but thought it looked rather ugly on the call side (especially in `create` where there would be a mixture of 
```
addSomething();
OS << "\n";
addSomethingElse();
```

and I preferred chain it using the stream operators. Yes, I know it’s not the best performance but I cared more about code readability than performance here because this part of the code likely isn’t very performance-critical.

Reply:
Yeah, that is indeed what I meant. I honestly don't mind the call side being split like that, but I can understand it being seen as ugly. Feel free to leave, just thought I'd mention it.

Reply:
I got curious now and tried the `add*` approach again. It does actually make all the helper methods slightly simpler and only `create` worse, so I changed it.